cli-tools/src/main/java/com/radixdlt/cli/GenerateGenesis.java

```diff
@@ -180,14 +183,45 @@ public static void main(String[] args) throws Exception {
         Base64.getEncoder().encodeToString(compressedGenesisData);
 
     if (validatorsCount > 0) {
+      // For some reason we use the presence of the validatorsCount command to decide whether we are
+      // running this for docker, or for an actual environment...
       System.out.format("export RADIXDLT_GENESIS_DATA=%s%n", compressedGenesisDataBase64);
     } else {
-      try (var writer = new BufferedWriter(new FileWriter("genesis.base64"))) {
-        writer.write(compressedGenesisDataBase64);
+      System.out.format(
+          "The base64-encoded genesis for use with RADIXDLT_GENESIS_DATA / network.genesis_data"
+              + " is:%n%s%n",
+          compressedGenesisDataBase64);
+      System.out.println();
+      var filePath = new File("genesis_data.bin").getAbsolutePath();
+      System.out.format(
+          "Also saving the genesis data file in binary format for use with"
+              + " RADIXDLT_GENESIS_DATA_FILE / network.genesis_data_file to: %s%n",
+          filePath);
+      try (var outputStream = new java.io.FileOutputStream(filePath)) {
+        outputStream.write(compressedGenesisData);
       }
     }
   }
 
+  private static Network parseNetwork(String commandLineValue) {
+    final var asName = Network.ofName(commandLineValue);
+    if (asName.isPresent()) {
+      return asName.get();
+    }
+    try {
+      final var asId = Network.ofId(Integer.parseInt(commandLineValue));
+      if (asId.isPresent()) {
+        return asId.get();
+      }
+      throw new RuntimeException("Was not a valid network id");
+    } catch (Exception ex) {
```

Comment:
hmm, something must have went wrong with the exceptions in this method (I mean: the `catch` 100% catches the `RuntimeException` thrown above)

Reply:
This was actually intentional - I wanted a wrapped error! Or more specifically, I wanted all the different errors that could be thrown to all be wrapped in an error with the same helpful error message.

Reply:
But based on your suggestion to use `Ints.tryParse()` I have gone full functional and flattened all this into 4 lines and avoided the gnarly errors 👍  :)